docs/proposals/images/controlplane/controlplane-init-7.plantuml

```diff
@@ -0,0 +1,55 @@
+@startuml
+title User scales down a KubeadmControlPlane
+actor User
+
+' -- GROUPS START ---
+
+box #lightgreen
+participant "API Server"
+end box
+
+box #violet
+participant "KubeadmControlPlane Controller"
+end box
+
+box #lightgreen
+participant "Workload Cluster API Server"
+end box
+
+' -- GROUPS END ---
+
+User->"API Server":kubectl scale cp/my-control-plane -replicas=1
+"API Server"-->>"KubeadmControlPlane Controller": KubeadmControlPlane Updated
+
+"KubeadmControlPlane Controller"-> "KubeadmControlPlane Controller":Enqueues KubeadmControlPlane Reconcile
+
+"KubeadmControlPlane Controller"-> "KubeadmControlPlane Controller":KubeadmControlPlane Controller Reconcile
+activate "KubeadmControlPlane Controller"
+
+note over "KubeadmControlPlane Controller": - ✅ KubeadmControlPlane.OwnerReferences \ncontains a Cluster
+
+"KubeadmControlPlane Controller"->"API Server": Get Cluster
+"KubeadmControlPlane Controller"<<--"API Server": Response
+
+note over "KubeadmControlPlane Controller": - ✅ Cluster.Status.InfrastructureReady is true\n- ✅ KubeadmControlPlane instance is valid
+
+"KubeadmControlPlane Controller"->"API Server": Get Machines maching label selector
+"KubeadmControlPlane Controller"<<--"API Server": Response
+
+note over "KubeadmControlPlane Controller": - Process for selecting a Machine to delete is TBD
+
+opt KubeadmControlPlane.Spec.Replicas >= 1, all existing Machines "Ready", num Machines > Replicas
+
+"KubeadmControlPlane Controller"->"API Server": Delete a Machine
```

Comment:
Not sure which way is correct. In the text "An upgrade will look like this" has the following order:
* Remove the replica’s etcd member from the etcd cluster
* Update the kubeadm generated config map
* Delete the marked controlplane machine

But here it's the other way around. Do we first have to remove the etcd member or delete the machine?

I think first removing the etcd member is correct.

Reply:
> I think first removing the etcd member is correct.

I agree.

(When we delete a machine, the etcd process on that machine is stopped. We should remove the etcd member before stopping the etcd process; if we remove it afterward, there is a period of time when the etcd cluster has a failed member, as opposed to being a smaller, but healthy, cluster.)

Reply:
delete-then-replace was the original text, but was considered unsuitable for single-node control planes in this model, so was swapped. maybe we need both to address the trade-offs.

Reply:
There is another important drawback to the delete first scenario...  What happens to the API Servers that are configured for and attempting to use the etcd peer that has been removed from the etcd cluster? Will it properly fall back to the other configured etcd peers? Can we guarantee that behavior in the future?

I worry a bit that if we remove the member prior to deleting the instance, then we are opening ourselves up to undefined behavior for a period of time vs operating in a degraded state for short period of time until we remove the member.

That said, in either scenario, I don't think there is a significant difference in the risk related to the number of additional failures that could be tolerated by the etcd cluster for the time period we are discussing.

In an ideal scenario, we would be able to trigger a `kubeadm reset` on the control plane Machine prior to shutting down, which would be a better way to reduce the risk window for this action, but that would require quite a bit of care/architecture to ensure that we could implement it across the various known infrastruture provider implementations.

Reply:
> I worry a bit that if we remove the member prior to deleting the instance, then we are opening ourselves up to undefined behavior for a period of time vs operating in a degraded state for short period of time until we remove the member.

By "undefined behavior," do you mean we don't know what will happen if a control plane replica, otherwise running and healthy, has its etcd member removed unilaterally and externally via the ecd members API call (as opposed to running `kubeadm reset` on the replica)?

(From the docs and my own experiments, when an etcd member is removed, the corresponding etcd process terminates as soon as it knows about the change.)

> In an ideal scenario, we would be able to trigger a kubeadm reset on the control plane Machine prior to shutting down, which would be a better way to reduce the risk window for this action, but that would require quite a bit of care/architecture to ensure that we could implement it across the various known infrastruture provider implementations.

I agree.

Reply:
Fixed the ordering in the image, thanks!